packages/calypso-e2e/src/lib/pages/signup-pick-plan-page.ts

```diff
@@ -25,7 +32,21 @@ export class SignupPickPlanPage {
 	 *
 	 * @param {Plans} name Name of the plan.
 	 */
-	async selectPlan( name: Plans ): Promise< void > {
-		await this.plansPage.selectPlan( name );
+	async selectPlan( name: Plans ): Promise< NewSiteDetails > {
+		const [ response ] = await Promise.all( [
+			this.page.waitForResponse( /.*sites\/new\?.*/ ),
+			this.plansPage.selectPlan( name ),
```

Comment:
Similar to what was done in [UserSignupPage](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/pages/user-signup-page.ts#L62-L83), the idea here is to intercept the response from the `/sites/new` endpoint. 

This request is triggered regardless of which plan the user selects in this screen:
![image](https://user-images.githubusercontent.com/6549265/170102810-256aa054-3396-4850-851a-6dda73f42b36.png)